Prox/Prox/Utilities/Future.swift

```diff
@@ -0,0 +1,18 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Deferred
+
+extension FutureProtocol {
+    /// Returns a Future resolved with the result of this Future, or nil
+    /// if this Future isn't resolved by the deadline.
+    func timeout(deadline: DispatchTime) -> Future<Value?> {
```

Comment:
I think it was crashes. From #29: "When Place creation returns a nil, the app is killed deliberately causing all hell to break loose. Make this case handle more gracefully somehow"

And from the commit (https://github.com/mozilla-mobile/prox/commit/404217f754a45cb73a21da3a6ecdd29771fd1d6f#diff-ef52f2a05b75e3e96af7cc9c6dc8245eR7):
+// This is an empty class that forms a base class for all places
+// It is done this way because:
+// 1. Deferred cannot handle optionals
+// 2. Deferred cannot handle Protocols, even if they are class protocols
+// 3. Deferred can only handle actual classes. Therefore if a Venue fails to initalize
+//    because it doesn't have all the info it needs to make itself
+//    we cannot return a nil through the deferred, we have to return a result object
+//    that can encapsulate the actual result of the call while still returning a 
+//    guarenteed value for Deferred